operators/advanced_network/Dockerfile

```diff
@@ -75,18 +75,91 @@ RUN --mount=type=bind,source=${HOLOSCAN_DEB_LOCAL_PATH},target=/tmp/holoscan.deb
 FROM hsdk-${HOLOSCAN_DEB_SRC} AS hsdk-installed
 
 # ==============================================================
-# common-deps: common dependencies for the advanced network lib
+# base-deps: base dependencies for the advanced network lib
 # ==============================================================
-FROM hsdk-installed AS common-deps
+FROM hsdk-installed AS base-deps
 
 ARG TARGETARCH
-ARG DOCA_VERSION=2.8.0
 ARG CACHEBUST=1
 ARG DEBIAN_FRONTEND=noninteractive
 
 WORKDIR /opt
 
-# Configure the DOCA APT repository
+# APT installs for base dependencies (no DOCA/GPUNetIO packages)
+# - libibverbs-dev, librdmacm-dev: RDMA/ibverbs support for Mellanox NICs
+# - libmlx5-1: Mellanox ConnectX driver
+# - ibverbs-utils: utilities
+# - python3-dev: for building python bindings
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        libibverbs-dev \
+        librdmacm-dev \
+        libmlx5-1 \
+        ibverbs-utils \
+        python3-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+# PIP installs
+# - pytest: test harness
+# - pyyaml: to parse yaml configs in tests
+# - scapy: for debugging and mocking network packets for tests
+RUN python3 -m pip install --no-cache-dir \
+        pytest \
+        pyyaml \
+        scapy
+
+# ==============================================================
+# dpdk-deps: Build upstream DPDK from source
+# ==============================================================
+FROM base-deps AS dpdk-deps
+
+ARG TARGETARCH
+# DPDK version to download and build
+ARG DPDK_VERSION=24.11.3
+ARG DPDK_BUILD_DIR=/opt/dpdk-build
+ARG DPDK_INSTALL_PREFIX=/usr/local
+ARG DOCA_VERSION=2.8.0
+ARG DEBIAN_FRONTEND=noninteractive
+
+
+# - infiniband-diags for IB diagnostics
+# - mft for Mellanox Flex Transport
+# - ninja-build: for cmake build
+# - pkgconf: to import dpdk in CMake
```

Comment:
aren't these 4 needed in base for all backends? the 2 first ones  as tools agnostic to the backend, the second ones as needed for cmake build in general